lib/take_away.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'send_sms'
+
+class TakeAway
+  attr_reader :menu
+
+  def initialize
+    @menu = []
+    @basket = []
+    @total = 0
+    import_file
+  end
+
+  def read_menu
+    menu.each do |item|
+      puts "#{item[:index]}. #{item[:item]}: £#{item[:amount]}"
+    end
+  end
+
+  def order(item_number, quantity = 1)
+    dish = menu[item_number - 1]
+    @basket << { item_number: item_number, quantity: quantity }
+    "You have added #{quantity} x #{dish[:item]} to your basket."
+  end
+
+  def basket_summary
+    @basket.each do |order|
+      dish = menu[order[:item_number] - 1]
+      item_total = convert(dish[:amount], order[:quantity])
+      @total += item_total.to_f
+
+      puts "#{order[:quantity]} x #{dish[:item]} = £#{item_total}"
+    end
+  end
+
+  def total
+    puts "Total: £#{convert(@total)}"
+    @total
+  end
+
+  def checkout(sum = 0)
+    raise "Incorrect amount, please enter the total." if sum != total
+
+    send_text
+  end
+
```

Comment:
I found it tricky to follow what was happening in each public method. It would be better to be more descriptive in a way faithful to what each method is trying to achieve, for example, what is "convert"? I know it is converting, but from what to what?